### PR TITLE
added endpoints for system metrics data

### PR DESCRIPTION
**Pull request by jenjenp**

### CODE REVIEW

File: services.yaml [hunk 1]
```diff
@@ -1884,6 +1886,130 @@ paths:
           $ref: "#/responses/UnauthorizedError"
         404:
           $ref: "#/responses/NotFoundError"
+#System Metrics
+  /services/{env_id}/metrics/system:
+    get:
+      tags:
+        - System Metrics
+      summary: "Get Metadatas"
+      description: "Get a list of all existing JVM Metrics"
```
Comment (nivstein) [inline services.yaml]: "Get a list of all available system metrics"

File: services.yaml [hunk 2]
```diff
@@ -1884,6 +1886,130 @@ paths:
           $ref: "#/responses/UnauthorizedError"
         404:
           $ref: "#/responses/NotFoundError"
+#System Metrics
+  /services/{env_id}/metrics/system:
+    get:
+      tags:
+        - System Metrics
+      summary: "Get Metadatas"
```
Comment (nivstein) [inline services.yaml]: "Get all metadata"

File: services.yaml [hunk 3]
```diff
@@ -1884,6 +1886,130 @@ paths:
...
+      description: "Get a list of all existing JVM Metrics"
+      parameters:
+        - in: path
+          name: env_id
+          description: Environment ID to use (e.g. S1234)
+          type: string
+          required: true
+      responses:
+        200:
+          description: List of metadatas
```
Comment (nivstein) [inline services.yaml]: "List of all system metrics' metadata"

File: services.yaml [hunk 4]
```diff
@@ -42,6 +42,8 @@ tags:
     description: Fetch and manipulate OverOps Timers - bottleneck detection and performance diagnosis
   - name: Publish Metrics
     description: Fetch and manipulate dynamic data exported to StatsD, which enables using a variety of third-party tools, providing control over application data from OverOps.
+  - name: System Metrics
+    description: Fetch JVM system metrics data
```
Comment (nivstein) [inline services.yaml]: "Fetch system metrics"

File: services.yaml [hunk 5]
```diff
@@ -1884,6 +1886,130 @@ paths:
...
+                  $ref: "#/definitions/SystemMetrics"
+        401:
+          $ref: "#/responses/UnauthorizedError"
+        404:
+          $ref: "#/responses/NotFoundError"
+  /services/{env_id}/metrics/system/{metric_name}:
+    get:
+      tags:
+        - System Metrics
+      summary: "Get Specific Metadata"
```
Comment (nivstein) [inline services.yaml]: "Get a specific metric's metadata"

File: services.yaml [hunk 6]
```diff
@@ -1884,6 +1886,130 @@ paths:
...
+      description: "Fetch info about specific Metadata"
```
Comment (nivstein) [inline services.yaml]: "Fetch info about a specific metric"

File: services.yaml [hunk 7]
```diff
@@ -1884,6 +1886,130 @@ paths:
...
+          type: string
+          required: true
+        - in: path
+          name: metric_name
+          description: Metric name to fetch
+          type: string
+          required: true
+      responses:
+        200:
+          description: List of metadatas
```
Comment (nivstein) [inline services.yaml]: "The requested metric's metadata"

File: services.yaml [hunk 8]
```diff
@@ -1884,6 +1886,130 @@ paths:
...
+            $ref: "#/definitions/SystemMetrics"
+        401:
+          $ref: "#/responses/UnauthorizedError"
+        404:
+          $ref: "#/responses/NotFoundError"
+  /services/{env_id}/metrics/system/{metric_name}/graph:
+    get:
+      tags:
+        - System Metrics
+      summary: Fetch Metrics data
```
Comment (nivstein) [inline services.yaml]: "Fetch metric graph/data"

File: services.yaml [hunk 9]
```diff
@@ -4213,6 +4339,33 @@ definitions:
...
+    type: object
+    properties:
+      type:
+        type: string
+        description: unit type
+        example: BYTES
+        enum: [NONE, CUSTOM, TIMES, NANOSECONDS, MICROSECONDS, MILLISECONDS, SECONDS, MINUTES, HOURS, DAYS, BYTES, KIBIBYTE, MEBIBYTE, GIBIBYTE]
+      suffix:
+        type: string
+        description: unit name for custom metadatas
```
Comment (nivstein) [inline services.yaml]: "singular suffix/descriptor for custom units"

File: services.yaml [hunk 10]
```diff
@@ -4213,6 +4339,33 @@ definitions:
...
+      plural_suffix:
+        type: string
+        description: unit plural name for custom metadatas
```
Comment (nivstein) [inline services.yaml]: "plural suffix/descriptor for custom units"

**Review by nivstein — CHANGES_REQUESTED** [review]

File: services.yaml (hunk 1 above)
Comment (jenjenp) [reply services.yaml]: fixed

**Review by nivstein — APPROVED** [review]

**Pull request merged**